CondCore/Utilities/plugins/BasicP_PayloadInspector.cc

```diff
@@ -91,6 +96,88 @@ namespace {
     }
   };
 
+  class BasicPayload_data6 : public cond::payloadInspector::PlotImage<cond::BasicPayload> {
+  public:
+    BasicPayload_data6() : cond::payloadInspector::PlotImage<cond::BasicPayload>( "Example delivery picture" ){
+      setSingleIov( true );
+    }
+
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+      auto iov = iovs.front();
+      std::shared_ptr<cond::BasicPayload> payload = fetchPayload( std::get<1>(iov) );
+  
+      double xmax(100.),ymax(100.);
+
+      TH2D *h2D = new TH2D("h2D","Example",100,0.,xmax,100,0.,ymax);
```

Comment:
not sure what I should provide? Your code owns this pointer, it needs to take care of it. Best practice is to use a smart pointer.